test/e2e_node/standalone_test.go

```diff
@@ -0,0 +1,189 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	"k8s.io/kubernetes/pkg/cluster/ports"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+)
+
+var _ = SIGDescribe("[Feature:StandaloneMode] ", func() {
+	f := framework.NewDefaultFramework("static-pod")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+	ginkgo.It("can create a static Pod ", func(ctx context.Context) {
+
+		var ns, podPath, staticPodName string
+
+		ns = f.Namespace.Name
+		staticPodName = "static-pod-" + string(uuid.NewUUID())
+
+		podPath = framework.TestContext.KubeletConfig.StaticPodPath
+
+		err := createBasicStaticPod(podPath, staticPodName, ns,
+			imageutils.GetE2EImage(imageutils.Nginx), v1.RestartPolicyAlways)
+		framework.ExpectNoError(err)
+
+		pod := pullPods(ctx, 1*time.Minute, 5*time.Second, staticPodName)
+
+		framework.ExpectEqual(pod.Status.Phase, v1.PodRunning)
+	})
+})
+
+func createBasicStaticPod(dir, name, namespace, image string, restart v1.RestartPolicy) error {
+	podSpec := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyAlways,
+			InitContainers: []v1.Container{
+				{
+					Name:  "init-1",
+					Image: busyboxImage,
+					Command: ExecCommand("init-1", execCommand{
+						Delay:    1,
+						ExitCode: 0,
+					}),
+				},
+			},
+			Containers: []v1.Container{
+				{
+					Name:  "regular1",
+					Image: busyboxImage,
+					Command: ExecCommand("regular1", execCommand{
+						Delay:    1000,
+						ExitCode: 0,
+					}),
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{
+							v1.ResourceMemory: resource.MustParse("15Mi"),
+						},
+						Limits: v1.ResourceList{
+							v1.ResourceMemory: resource.MustParse("15Mi"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	podYaml, err := kubeadmutil.MarshalToYaml(podSpec, v1.SchemeGroupVersion)
+	if err != nil {
+		return err
+	}
+
+	file := staticPodPath(dir, name, namespace)
+
+	f, err := os.OpenFile(file, os.O_RDWR|os.O_TRUNC|os.O_CREATE, 0666)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	_, err = f.Write(podYaml)
```

Comment:
Do we want to delete this file when the test is done?

Reply:
test runs will not interfere with each other as we use randomized folder name for static pods. But yes, cleaning up in this test would make sense.